### PR TITLE
fix: transfer amount

### LINKED ISSUE #6711: Show correct value with decimals in new Transfer page and allow user to send less than 0.01 KSM
### What happened?

Modify transfer page logic in order to show the correct total amount of USD prior to transfer and also allow users to send amounts smaller than 0.01 KSM using the page. 

Related to #6695 

Comments: 
https://github.com/kodadot/nft-gallery/pull/6695#issuecomment-1677683558
https://github.com/kodadot/nft-gallery/pull/6695#issuecomment-1677809528

by @Jarsen136 :
> If you input 0.05 usd, it would be converted to 0.01 ksm. Eventually, every address would receive 0.01 ksm($0.23). It's correct, and It also follows the transfer logic as before. 
> 

- **Problem related to float and integer variables:** 

260531409-34a75953-be41-4968-aed6-111481ff5f57

260530369-df7f00c5-8937-4bb4-9a21-c9d3e6ac42a8

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #6711
- [x] Set the value of USD with two decimal places and the number of tokens with four decimal places. 

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [x] My fix has changed UI

Old design of transfer

image

Redesigned transfer page

image

I have made a test transfer with 0.0022KSM($0.05)

image

tx: https://kusama.subscan.io/extrinsic/19246871-3

### <samp>🤖 Generated by Copilot at 75a6ec1</samp>

This pull request improves the calculation and display of USD values for different tokens in the top collections and transfer components. It also refactors some utility functions and hooks to enhance code quality and consistency.

### <samp>🤖 Generated by Copilot at 75a6ec1</samp>

> _To calculate tokens in bucks_
> _They refactored some utility funcs_
> _They also replaced_
> _The hooks that they faced_
> _And used getters instead of state flux_

### CODE REVIEW

File: utils/calculation.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-export function calculateUsdFromKsm(ksm: number, price: number): number {
-  return Number(Math.ceil(ksm * price))
+export function calculateUsdFromToken(amount: number, price: number): number {
```
[the remark this answers is not in the document]
Comment (Jarsen136) [reply utils/calculation.ts:1]: I could not find the similar one : )

**Comment by prury** [comment]
looking really nice! good job! 
https://kusama.subscan.io/tx/0x9b8fba6717b47ef5441f416dc4578df70bb18ffa1a8744da24593b5ebb743862
https://kusama.subscan.io/tx/0x7c498260e7f902116ab07a964f65b39f4c2faaf4f777806539cb3f1bd8f842f9

**Review by roiLeo — APPROVED** [review]

File: components/landing/topCollections/TopCollectionsItem.vue [hunk 2]
```diff
@@ -68,7 +68,7 @@
 
 <script lang="ts" setup>
 import { TimeRange } from '@/components/series/types'
-import { calculateUsdFromKsm } from '~~/utils/calculation'
+import { calculateUsdFromToken } from '~~/utils/calculation'
```
Comment (preschian) [inline components/landing/topCollections/TopCollectionsItem.vue:71]:
```suggestion
import { calculateUsdFromToken } from '@/utils/calculation'
```

**Review by preschian — APPROVED** [review]
otherwise lgtm

**Comment by vikiival** [comment]
Thanks for followup!

**Pull request merged**